explore/src/main/scala/explore/targeteditor/AladinContainer.scala

```diff
@@ -214,51 +101,98 @@ object AladinContainer {
+                val g        = document.createElement("div")
+                g.classList.add("aladin-visualization")
+                visualization.geometryForAladin(svg.value,
+                                                g,
+                                                size,
+                                                pixelScale,
+                                                GmosGeometry.ScaleFactor,
+                                                (x, y)
+                )
+                // Switch the visibility
+                toggleVisibility(g, "#science-ccd polygon", props.options.fov)
+                toggleVisibility(g, "#science-ccd-offset polygon", props.options.offsets)
+                toggleVisibility(g, "#patrol-field", props.options.guiding)
+                toggleVisibility(g, "#probe", props.options.probe)
+                div.appendChild(g)
+                ()
+              case _            =>
+            }
+
+        def updateVisualization(v: JsAladin): Callback = {
+          val size = Size(v.getParentDiv().clientHeight, v.getParentDiv().clientWidth)
+          val div  = v.getParentDiv()
+          renderVisualization(div, size, v.pixelScale)
         }
-      )
 
-    def recalculateView =
-      aladinRef.get.asCBO.flatMapCB { r =>
-        r.backend.runOnAladinCB { v =>
-          updateSvgState(v).flatMap(s =>
-            r.backend.recalculateView *> r.backend.runOnAladinCB(updateVisualization(s))
-          )
-        }
-      }
-  }
+        def onZoom = (v: Fov) =>
+          props.fov.set(v) *>
+            props.updateFov(v) *> recalculateView
+
+        def recalculateView =
+          aladinRef.get.asCBO.flatMapCB { r =>
+            r.backend.recalculateView *> r.backend.runOnAladinCB(updateVisualization)
+          }
+
+        def includeSvg(v: JsAladin): Callback =
+          v.onFullScreenToggle(recalculateView) *> // re render on screen toggle
+            v.onZoom(onZoom) *>                    // re render on zoom
+            v.onPositionChanged(onPositionChanged(v)) *>
+            v.onMouseMove(s =>
+              props
+                .updateMouseCoordinates(Coordinates(s.ra, s.dec))
+                .debounce(200.millis)
+                .void
+            )
 
-  val component =
-    ScalaComponent
-      .builder[Props]
-      .initialState(State.Zero)
-      .renderBackend[Backend]
-      .componentDidUpdate(_.backend.recalculateView.toCallback)
-      .configure(Reusability.shouldComponentUpdate)
-      .build
+        <.div(
+          ExploreStyles.AladinContainerBody,
+          // This is a bit tricky. Sometimes the height can be 0, this happens if
+          // during a second render. If we let the height to be zero, aladin will
+          // take it as 1 and miss calculate the amount of tiles needed.
+          // This can endup requesting a large amount of tiles and would
+          // freeze explore
+          if (resize.height.exists(_ > 0))
```

Comment:
Good detective work! Is there any chance the value could actually be 1 and cause the problem? Is 1 a magic number in aladin?

Reply:
Thank you for tracking down and fixing this